index.js

```diff
@@ -147,7 +164,25 @@ const promiseSerial = funcs => {
   }, Promise.resolve([]));
 };
 
-const backupDocument = (
+const getTransformedDocument = async (
+  collectionPath: Array,
+  document: Object
+) => {
+  const backupDocObject = constructDocumentObjectToBackup(document.data());
+  const obj = !!executeTranformFn
```

Comment:
I think it is often necessary to know the type of each of the fields, probably we will want to do different actions depending on whether the field is of a certain type
for example: if the user needs to change a string id for a reference, he should ask if the field in question is of type string before applying the transformation function (I think the case below would also apply to this)

I still can not imagine how removing the backup format can simplify things... but I think there may be certain drawbacks with the different types.
for example, in the hypothetical case in which the user wants to change the backup of a geopoint, the value when one reads it is `{" _latitude ": 1," _longitude ": -1}` then from the backup data it would not be possible to know with what type should be restored, if it should be an object with that fields or a `geopoint` type

Reply:
Yes. You are right. I now recall that types.js isGeopoint and isDocumentReference require in-memory object from firestore ask to determine the proper type. So let's disregard this question!